SharedProjects/BabylonExport.Entities/BabylonMesh.cs

```diff
@@ -5,12 +5,31 @@
 
 namespace BabylonExport.Entities
 {
-    
+    public interface IBabylonMeshData
+    {
+        float[] positions { get; set; }
+        float[] normals { get; set; }
+        float[] tangents { get; set; }
+        float[] uvs { get; set; }
+        float[] uvs2 { get; set; }
+        float[] uvs3 { get; set; }
+        float[] uvs4 { get; set; }
+        float[] uvs5 { get; set; }
+        float[] uvs6 { get; set; }
+        float[] colors { get; set; }
+        int[] matricesIndices { get; set; }
+        float[] matricesWeights { get; set; }
+        int[] indices { get; set; }
+    }
+
     [DataContract]
-    public class BabylonMesh : BabylonAbstractMesh
+    public class BabylonMesh : BabylonAbstractMesh, IBabylonMeshData
     {
         [DataMember]
         public string materialId { get; set; }
+        
+        [DataMember]
+        public string geometryId { get; set; }
```

Comment:
How are we serializing this to glTF?

Reply:
GLTFExporter.Mesh is in charge. 
I choose to abstract the geometry throught an interface 
```
    public interface IBabylonMeshData
    {
        float[] positions { get; set; }
        float[] normals { get; set; }
        float[] tangents { get; set; }
        float[] uvs { get; set; }
        float[] uvs2 { get; set; }
        float[] uvs3 { get; set; }
        float[] uvs4 { get; set; }
        float[] uvs5 { get; set; }
        float[] uvs6 { get; set; }
        float[] colors { get; set; }
        int[] matricesIndices { get; set; }
        float[] matricesWeights { get; set; }
        int[] indices { get; set; }
    }
```
An then it's kind of transparent for the GLTF serializer to work with a mesh data OR geometry. 
This is also partially respond to your previous comment.

Reply:
GLTFExporter.Mesh is in charge. 
I choose to abstract the geometry throught an interface 
```
    public interface IBabylonMeshData
    {
        float[] positions { get; set; }
        float[] normals { get; set; }
        float[] tangents { get; set; }
        float[] uvs { get; set; }
        float[] uvs2 { get; set; }
        float[] uvs3 { get; set; }
        float[] uvs4 { get; set; }
        float[] uvs5 { get; set; }
        float[] uvs6 { get; set; }
        float[] colors { get; set; }
        int[] matricesIndices { get; set; }
        float[] matricesWeights { get; set; }
        int[] indices { get; set; }
    }
```
An then it's kind of transparent for the GLTF serializer to work with a mesh data OR geometry. 
This is also partially respond to your previous comment.

Reply:
I'm not sure if i'm being clear here.

Imagine we have two meshes, one original and a clone of that original mesh in scene:

when serialized both will be serialized as IBabylonMeshData, however the fields in the clone are now null. Currently we are adding no additional logic to the glTF serializer to handle serialization of meshes with GeometryData, so I'm concerned that the second mesh may not be properly serialized, as we will be using those nulled fields.